hoomd/md/pytest/test_custom_force.py

```diff
@@ -0,0 +1,323 @@
+import pytest
+import numpy as np
+import numpy.testing as npt
+
+try:
+    import cupy
+    CUPY_IMPORTED = True
+except ImportError:
+    CUPY_IMPORTED = False
+
+import hoomd
+from hoomd import md
+
+
+class MyForceCPU(md.force.Custom):
+
+    def __init__(self):
+        super().__init__()
+
+    def set_forces(self, timestep):
+        with self.cpu_local_force_arrays as arrays:
+            arrays.force = -5
+            arrays.potential_energy = 37
+            arrays.torque = 23
+            arrays.virial = np.arange(6)[None, :]
+
+
+class MyForceGPU(md.force.Custom):
+
+    def __init__(self):
+        super().__init__()
+
+    def set_forces(self, timestep):
+        with self.gpu_local_force_arrays as arrays:
+            arrays.force = -5
+            arrays.potential_energy = 37
+            arrays.torque = 23
+            # cupy won't let me use two indices on arrays of length 0
+            if len(arrays.virial) > 0:
```

Comment:
Maybe an assert would make sense here? Since we apparently need cupy to run this test anyways and there's not another case I can think of that would trigger it.

Reply:
In some of the tests with MPI, there are 0 particles on one of the MPI ranks. If we put an assert here, the test will fail on that rank.